arbitrum-docs/stylus/stylus-quickstart.md

```diff
@@ -11,209 +11,242 @@ import PublicPreviewBannerPartial from '../partials/_public-preview-banner-parti
 
-If the command above fails, you'll see detailed information about why your WASM will be rejected:
 
-```
+If the command above fails, you'll see detailed information about why your contract would be rejected:
+
+```shell
 Reading WASM file at bad-export.wat
 Compressed WASM size: 55 B
-Stylus checks failed: program predeployment check failed when checking against
+Stylus checks failed: program pre-deployment check failed when checking against
 ARB_WASM_ADDRESS 0x0000…0071: (code: -32000, message: program activation failed: failed to parse program)
 
 Caused by:
     binary exports reserved symbol stylus_ink_left
 
 Location:
     prover/src/binary.rs:493:9, data: None)
-
 ```
 
-To read more about what counts as valid vs. invalid user WASM programs, see [VALID_WASM](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md). If your program succeeds, you'll see the following message:
+When a program fails the check, it means that there was a problem with the WASM binary generated by `cargo-stylus`.
+To read more about what counts as valid vs. invalid user WASM contracts, you can visit the [Invalid Stylus WASM Contracts explainer](https://github.com/OffchainLabs/cargo-stylus/blob/main/main/VALID_WASM.md).
 
-```
+If your program succeeds, you'll see something like this:
+
+```shell
 Finished release [optimized] target(s) in 1.88s
 Reading WASM file at hello-stylus/target/wasm32-unknown-unknown/release/hello-stylus.wasm
 Compressed WASM size: 3 KB
 Program succeeded Stylus onchain activation checks with Stylus version: 1
 ```
 
-Note that running `cargo stylus check` may take a few minutes, especially on a fresh build. If you're running the check or deploy commands, a message at the start will inform you that this process might take some time. For faster checks, you can use the `--no-verify` option. Also, please ensure that Docker is up and running, as these commands are required to be executed properly.
+Note that running `cargo stylus check` may take a few minutes, especially if you're checking a contract for the first time. 
+You can use the `-no-verify` option for faster checks.
 
-Once you're ready to deploy your program onchain, you can use the `cargo stylus deploy` subcommand as follows. First, we can estimate the gas required to perform our deployment with:
+See `cargo stylus check --help` for more options.
 
-```
-cargo stylus deploy \
-  --private-key-path=<PRIVKEY_FILE_PATH> \
-  --estimate-gas-only
-```
+## Deploying your contract
 
-and see:
+Once you're ready to deploy your program onchain, `cargo stylus deploy` will help you with the estimation of the deployment's gas cost, as well as with the deployment itself.
 
+### Estimating gas
+
+
+First, you can estimate the gas required to perform your contract's deployment by running:
+
+```shell
+cargo stylus deploy \
+  --private-key-path=PRIV_KEY_PATH \
```

Comment:
I think this should be `--private-key-path=$PRIV_KEY_PATH` ? (notice the `$` symbol)

I like this approach (using the path to the private key file, I'll edit the previous comment)💪

Reply:
Also, above we used PRIVATE_KEY_PATH instead of PRIV_KEY_PATH

Reply:
Both of the comments above are still relevant

Reply:
Fixed.